src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -2303,3 +2303,19 @@ class DropTriggerStatementSegment(BaseSegment):
         Ref("TableReferenceSegment"),
         OneOf("CASCADE", "RESTRICT", optional=True),
     )
+
+
+@postgres_dialect.segment(replace=True)
+class InsertStatementSegment(BaseSegment):
+    """An `INSERT` statement."""
+
+    type = "insert_statement"
+    match_grammar = StartsWith("INSERT")
+    parse_grammar = Sequence(
+        "INSERT",
+        "INTO",
+        Ref("TableReferenceSegment"),
+        Ref("BracketedColumnReferenceListGrammar", optional=True),
+        Sequence("OVERRIDING", OneOf("SYSTEM", "USER"), "VALUE", optional=True),
+        Ref("SelectableGrammar"),
+    )
```

Comment:
```SQL
@ansi_dialect.segment()
class InsertStatementSegment(BaseSegment):
    """A `INSERT` statement."""

    type = "insert_statement"
    match_grammar = StartsWith("INSERT")
    parse_grammar = Sequence(
        "INSERT",
        Ref.keyword("OVERWRITE", optional=True),  # Maybe this is just snowflake?
        Ref.keyword("INTO", optional=True),
        Ref("TableReferenceSegment"),
        Ref("BracketedColumnReferenceListGrammar", optional=True),
        Ref("SelectableGrammar"),
    )
```
The current ANSI implementation for the reviewers convenience.

Reply:
A couple questions on the ANSI version:
1. I couldn't find any evidence that the `OVERWRITE` keyword should be in any other dialect except Snowflake (as the comment says). This should be moved to Snowflake dialect. Happy to do this as part of this PR or make a seperate one.
2. I'm not aware of any dialect (certainly ansi/postgres/snowflake/big query) where the `INTO` keyword is optional? Is there an example of this and if so it should only be made optional in the relevant dialect.

Reply:
I have corrected the 2 cases above in the postgres dialect but feel they should be updated in ANSI.

Reply:
So, our ANSI isn't strictly ANSI, it's kind of ANSI or stuff that is easier to just have in ANSI. If you can confirm the INTO isn't optional in any dialect (I'd check SQLite because it's weird), and it doesn't break tests, it's worth doing. Probably not worth removing overwrite from ANSI at this juncture

Reply:
makes sense, I figured that was the case. Let me just take a quick look at the optional INTO.

Reply:
Okay so the 2 culprits I can find are:
Big Query
![image](https://user-images.githubusercontent.com/80432516/141019877-97f053c4-7381-4b43-be60-cf736d2512df.png)

Hive
![image](https://user-images.githubusercontent.com/80432516/141020154-3f2eb06f-6388-4edf-bb43-0bfa18cf5c43.png)
(overwrite as well!)

Reply:
@WittierDinosaur Hive already has a complete implementation of it's insert grammar (including it's optional INTO) so I've removed the optional INTO in ANSI and moved to big query. No tests are failing and I've also added an extra unit test to big query as well to check the optional INTO in big query.

Reply:
Good effort - LGTM